### PR TITLE
Add missing padding needded for alignment

### LINKED ISSUE #2857: Unexpected r value in box conversion
I tried this code:

```rust
#[kani::proof]
fn example() {
   let secret_key =
       p384::SecretKey::from_slice(&[0]).expect("Failed to import key.");

   let public_key = secret_key.public_key().to_sec1_bytes();  // without to_sec1_bytes(), build will be successful.
}

fn main() {
}
```

with `Cargo.toml`
```
[package]
name = "example"
version = "0.1.0"
edition = "2021"

[dependencies]
p384 = { version = "*", default-features = false, features = ["alloc", "ecdsa"] }
```

using the following command line invocation:

```
$ cargo kani
```

with Kani version: 0.40 (341836c40a737c05275790ff5d6f9c1f13e71b7b)

I expected to see this happen: build successful

Instead, this happened: r value interpretation seems to be different. 

```
thread 'rustc' panicked at kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1289:9:
assertion `left == right` failed
  left: 56
 right: 49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Kani unexpectedly panicked during compilation.
Please file an issue here: https://github.com/model-checking/kani/issues/new?labels=bug&template=bug_report.md

[Kani] current codegen item: codegen_function: std::result::Result::<p384::elliptic_curve::sec1::Tag, sec1::error::Error>::expect
_RNvMNtCsHPzDkKRnVq_4core6resultINtB2_6ResultNtNtCsh3yxSpuaCtL_4sec15point3TagNtNtBK_5error5ErrorE6expectCs8ec89RYc48b_7example
[Kani] current codegen location: Loc { file: "/home/changho/.rustup/toolchains/nightly-2023-10-31-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs", function: None, start_line: 1028, start_col: Some(5), end_line: 1030, end_col: Some(23) }
error: internal compiler error: Kani unexpectedly panicked at panicked at kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1289:9:
                                assertion `left == right` failed
                                  left: 56
                                 right: 49.

error: could not compile `example` (bin "example"); 2 warnings emitted
error: Failed to compile `example` due to an internal compiler error.: error: internal compiler error: Kani unexpectedly panicked at panicked at kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1289:9:
                                assertion `left == right` failed
                                  left: 56
                                 right: 49.
```

[backtrace]
```
   0: rust_begin_unwind
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/core/src/panicking.rs:72:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/core/src/panicking.rs:279:5
   4: kani_compiler::codegen_cprover_gotoc::codegen::rvalue::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::check_vtable_size
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1289:9
   5: kani_compiler::codegen_cprover_gotoc::codegen::rvalue::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_vtable::{{closure}}
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1354:35
   6: kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx::ensure_global_var
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/context/goto_ctx.rs:223:33
   7: kani_compiler::codegen_cprover_gotoc::codegen::rvalue::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_vtable
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1337:9
   8: kani_compiler::codegen_cprover_gotoc::codegen::rvalue::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_cast_to_fat_pointer
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1443:30
   9: kani_compiler::codegen_cprover_gotoc::codegen::rvalue::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_unsized_cast
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1087:28
  10: kani_compiler::codegen_cprover_gotoc::codegen::rvalue::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_pointer_cast
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1056:17
  11: kani_compiler::codegen_cprover_gotoc::codegen::rvalue::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_rvalue
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:710:17
  12: kani_compiler::codegen_cprover_gotoc::codegen::statement::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_statement
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs:54:33
  13: kani_compiler::codegen_cprover_gotoc::codegen::block::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_block
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs:57:32
  14: kani_compiler::codegen_cprover_gotoc::codegen::function::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_function::{{closure}}
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs:84:57
  15: core::iter::traits::iterator::Iterator::for_each::call::{{closure}}
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/core/src/iter/traits/iterator.rs:855:29
  16: core::iter::adapters::map::map_fold::{{closure}}
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/core/src/iter/adapters/map.rs:84:21
  17: <core::slice::iter::Iter<T> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/core/src/slice/iter/macros.rs:232:27
  18: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/core/src/iter/adapters/map.rs:124:9
  19: core::iter::traits::iterator::Iterator::for_each
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/core/src/iter/traits/iterator.rs:858:9
  20: kani_compiler::codegen_cprover_gotoc::codegen::function::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_function
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs:84:13
  21: kani_compiler::codegen_cprover_gotoc::compiler_interface::GotocCodegenBackend::codegen_items::{{closure}}::{{closure}}
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs:128:39
  22: kani_compiler::codegen_cprover_gotoc::utils::debug::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::call_with_panic_debug_info::{{closure}}
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/utils/debug.rs:68:13
  23: std::thread::local::LocalKey<T>::try_with
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/std/src/thread/local.rs:270:16
  24: std::thread::local::LocalKey<T>::with
             at /rustc/31bc7e2c47e82798a392c770611975a6883132c8/library/std/src/thread/local.rs:246:9
  25: kani_compiler::codegen_cprover_gotoc::utils::debug::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::call_with_panic_debug_info
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/utils/debug.rs:65:9
  26: kani_compiler::codegen_cprover_gotoc::compiler_interface::GotocCodegenBackend::codegen_items::{{closure}}
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs:127:29
  27: kani_compiler::codegen_cprover_gotoc::compiler_interface::with_timer
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs:748:15
  28: kani_compiler::codegen_cprover_gotoc::compiler_interface::GotocCodegenBackend::codegen_items
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs:97:9
  29: <kani_compiler::codegen_cprover_gotoc::compiler_interface::GotocCodegenBackend as rustc_codegen_ssa::traits::backend::CodegenBackend>::codegen_crate
             at /home/changho/zpzigi754/kani/kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs:239:25
  30: rustc_interface::passes::start_codegen
  31: <rustc_interface::queries::Queries>::ongoing_codegen
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

Kani unexpectedly panicked during compilation.
Please file an issue here: https://github.com/model-checking/kani/issues/new?labels=bug&template=bug_report.md

[Kani] current codegen item: codegen_function: std::result::Result::<p384::elliptic_curve::sec1::Tag, sec1::error::Error>::expect
_RNvMNtCsHPzDkKRnVq_4core6resultINtB2_6ResultNtNtCsh3yxSpuaCtL_4sec15point3TagNtNtBK_5error5ErrorE6expectCs8ec89RYc48b_7example
[Kani] current codegen location: Loc { file: "/home/changho/.rustup/toolchains/nightly-2023-10-31-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs", function: None, start_line: 1028, start_col: Some(5), end_line: 1030, end_col: Some(23) }
error: could not compile `example` (bin "example"); 2 warnings emitted
```

The definition for `to_sec1_bytes()` is as follows.
```rust
    #[cfg(all(feature = "alloc", feature = "sec1"))]
    pub fn to_sec1_bytes(&self) -> Box<[u8]>
    where
        C: PointCompression,
        AffinePoint<C>: FromEncodedPoint<C> + ToEncodedPoint<C>,
        FieldBytesSize<C>: ModulusSize,
    {
        EncodedPoint::<C>::from(self).to_bytes()
    }
```

### LINKED ISSUE #3318: Kani panicked in codegen/rvalue
```
thread 'rustc' panicked at kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1317:9:error: internal compiler error: Kani unexpectedly panicked at panicked at kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1317:9:
                                assertion `left == right` failed
                                  left: 48
                                 right: 41.


assertion `left == right` failed
  left: 48
 right: 41
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Kani unexpectedly panicked during compilation.
Please file an issue here: https://github.com/model-checking/kani/issues/new?labels=bug&template=bug_report.md

[Kani] current codegen item: codegen_function: <der::error::Error as std::fmt::Debug>::fmt
_RNvXs9_NtCslefA7hYQHvl_3der5errorNtB5_5ErrorNtNtCscBmSNMcqgUz_4core3fmt5Debug3fmtCsg8FMlfzrjoY_7rslip10
[Kani] current codegen location: Loc { file: "$HOME/.cargo/registry/src/index.crates.io-6f17d22bba15001f/der-0.7.9/src/error.rs", function: None, start_line: 18, start_col: Some(23), end_line: 18, end_col: Some(28) }
error: could not compile `rslip10` (lib test)
error: Failed to compile `rslip10` due to an internal compiler error.: error: internal compiler error: Kani unexpectedly panicked at panicked at kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:1317:9:
                                assertion `left == right` failed
                                  left: 48
                                 right: 41.
```

This is when running `cargo kani --package my_package --tests` and, evidently, kani has an issue with the latest `der` crate.

### PR DESCRIPTION
**Pull request by zhassan-aws**
This PR adds extra padding for enums when needed for alignment. The missing alignment was exposed in #2857 where the layout size from MIR for `der::error::ErrorKind` was 48 bytes, and from `codegen_enum` was only 41 bytes. The difference turned out to be caused by an 8-byte alignment that the compiler uses for this enum.

Resolves #2857 
Resolves #3318 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by tautschnig — APPROVED** [review]
Thank you!

File: kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs [hunk 1]
```diff
@@ -1240,8 +1239,17 @@ impl<'tcx> GotocCtx<'tcx> {
                                         variants,
                                         initial_offset,
                                     )
-                                }),
-                            ));
+                                });
+                            let cases_size = Size::from_bytes(cases.sizeof(&gcx.symbol_table));
```
Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs:1243]: Could we get that information from rustc LayoutOf?

Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs:1243]: Could we get that information from rustc LayoutOf?

Comment (zhassan-aws) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs]: Done. Not sure if the way I implemented it is the best though.

**Review by celinval — APPROVED** [review]
Thanks!

**Pull request merged**